feat(files): Rename files

lgmarchi opened the pull request:
<!--  Thanks for sending a pull request!-->

### What this PR does 📖
### 1. Add possibility to rename a file 
a. Moved Storage to ui_utils, to be possible to use inside File component, and with it needed to remove Eq derive from props 
b. File component now has some use_state and use_ref variables, because we need to rename it and update the component

### 2. New menu to options is now with context menu using mouse right click 
a. Old menu as just for test, this one is official, but download option not working yet

### 3. A key for file component was necessary to help Dioxus doesn't lost reference of items 
a. Without this key, ContextMenu lost reference and when upload a new file it comes from same name as the first one

### 4. Remove filter on file dialog to be possible upload any kind of file 
a. No big changes, just possible to upload other kind of files now

### Which issue(s) this PR fixes 🔨
- Resolve #376 
#377 
<!--Add the ticket Github number such as #Resolve #001 to automatically link the PR to the issue-->

### Special notes for reviewers 🗒️
@phillsatellite Is possible to test rename with no problems, but it is missing an update yet from warp to not allow rename the same file in same directory with same name, the PR is already open, just need to be merged and this PR will be updated. 



### Additional comments 🎤

lgmarchi committed:
Storage to utils, allow to use in ui_kit

lgmarchi committed:
just style settings

lgmarchi committed:
Show temp menu to test

lgmarchi committed:
Add possibility to rename files

lgmarchi committed:
Remove filter to upload any kind of file

lgmarchi committed:
Merge branch 'dev' of https://github.com/Satellite-im/Uplink-Desktop into rename_files

lgmarchi committed:
Move context menu to File entity

lgmarchi committed:
Change complete file name to use_ref

lgmarchi committed:
Merge branch 'dev' of https://github.com/Satellite-im/Uplink-Desktop into rename_files

lgmarchi committed:
Add key to Files component to avoid lost reference

lgmarchi committed:
Change UI for input field

dariusc93 committed:
chore: Update dependency

lgmarchi committed:
Merge branch 'chore/update-r5' of https://github.com/Satellite-im/Uplink into rename_files

lgmarchi committed:
Merge branch 'dev' of https://github.com/Satellite-im/Uplink into rename_files

phillsatellite commented:
Hi Lucas! Found a few things on here while testing,

1. When the User clicks rename, they should be able to start typing without having to click into the textbox first 

https://user-images.githubusercontent.com/93608357/204064974-e6b0f827-dd36-4d75-bf8e-34fa856006a0.mov

2. When the user clicks out of the textbox without typing anything or saving we should revert the file name back to what they had before

https://user-images.githubusercontent.com/93608357/204065063-84ae35c5-5b49-4289-9b42-98c88f46d477.mov

3. If User rename's a file to the same name of a file they already have uploaded, they will not be able to delete either of them 

https://user-images.githubusercontent.com/93608357/204065110-0a41478e-dda8-4e40-ac55-4d3edd76ecbe.mov

lgmarchi committed:
chore: Update cargo.toml with last dev from warp

dariusc93 reviewed (dismissed):
I would suggest avoiding updating warp commit since i usually do testing when I made do a pr to update it here in uplink.

lgmarchi committed:
refactor(file): Focus on input element when start to rename file

lgmarchi committed:
refactor(Files): Temp solution to exit edit name when click out file or move pointer outside file browser

InfamousVague reviewed (approved):
This solution should have a // TODO: comment added to clean it up I think there is a better way to do this, but if this works as expected with QA it looks good to me for now.

lgmarchi committed:
Merge branch 'dev' of https://github.com/Satellite-im/Uplink into rename_files

lgmarchi committed:
refactor(file): Use javascript to hide and show edit file name

lgmarchi committed:
refactor(file): clear value on input element if it closes or open

dariusc93 commented on src/ui_kit/src/file/mod.rs:
```diff
@@ -19,19 +26,130 @@ pub fn File<'a>(cx: Scope<'a, Props>) -> Element<'a> {
+                                    icon: Shape::PencilSquare,
+                                    onpressed: move |_| {
+                                        // Use js script to show edit file name element
+                                        use_eval(&cx)(&show_edit_name_script);
```
Something to becareful about when doing it this way as replacing a section of the script while loading it in (as you are doing above) and evaluating it here could open possibly of an xss (though info wont be impacted but could still open holes if one is not careful with what its being replaced with, etc. Did leave a suggestion just adding a comment for it. Also using this function could panic if the js is invalid so something to becareful of too. 

```suggestion
                                        //TODO: Investigate in a way to replace use_eval in the future
                                        use_eval(&cx)(&show_edit_name_script);
```

lgmarchi replied:
Thanks Darius, I added TODO, something important to care. I used use_eval as recommendation from dioxus lab devs, maybe in master, should have better ways to run a js script.

I tested a invalid js, and the app didn't have panic, just not worked. 

https://user-images.githubusercontent.com/63157656/204306638-7b350a59-8dd7-4f2a-ae60-fe30a1856e34.mov

dariusc93 commented on src/ui_kit/src/file/mod.rs:
```diff
@@ -19,19 +26,130 @@ pub fn File<'a>(cx: Scope<'a, Props>) -> Element<'a> {
+                                    let file_extension = cx.props.kind.clone();
+                                    let new_file_name = val.read();
+                                    // Use js script to hide edit file name element
+                                    use_eval(&cx)(&hide_edit_name_script);
```
```suggestion
                                    //TODO: Investigate in a way to replace use_eval in the future
                                    use_eval(&cx)(&hide_edit_name_script);
```

dariusc93 reviewed (approved):
Looks good to me so far. Did leave a suggestion below though

lgmarchi committed:
refactor(file): Add TODO to change use_eval in future

- Remove event listener when click out file element
- Keep input focus if click on file element

lgmarchi committed:
Merge branch 'dev' of https://github.com/Satellite-im/Uplink into rename_files
